stl/inc/xatomic.h

```diff
@@ -56,6 +56,15 @@ _STL_DISABLE_CLANG_WARNINGS
 _STD_BEGIN
 
 // ENUM memory_order
+#if _HAS_CXX20
+enum class memory_order : unsigned int { relaxed, consume, acquire, release, acq_rel, seq_cst };

```

Comment:
I’d be more comfortable with a Tentatively Ready vote, but it’s “just” restoring something that was possible in C++17. My main concern is any interaction with `using enum`, has the LWG thought about that?

Reply:
> My main concern is any interaction with `using enum`, has the LWG thought about that?

I doubt it, given how new `using enum` is. I don't see it being problematic except for code that's also `using namespace std;` and refers to the "old" names unqualified:
```c++
using namespace std;
using enum memory_order;

int foo(atomic<int>& a) {
  return 42 + a.load(memory_order_relaxed); // Error: memory_order_relaxed is ambiguous
}
```
which seems a peculiar choice to me.

Reply:
Particularly given that the using enum should have no effect since we have the inline constexpr fallbacks.

>I’d be more comfortable with a Tentatively Ready vote,

It has more than 5 P0s on the LWG reflector, it just hasn't been updated by our lord and savior Daniel Krügler yet.

Reply:
It looks like the issue won't be accepted as P0, but it would be conformant to provide the enumerators and mark them as deprecated to encourage migration. We should decide what to do here.

Reply:
AFAIK the only objection to P0 was mine, and the objection was "I want to see evidence that users actually did this before we create a thing users have to migrate off of", evidence that users were doing it was provided in the reflector thread, and I withdrew my objection. /cc @Dani-Hub

Reply:
One person voted P4 only, and another commented that LEWG should approve such a change (I am avoiding citing specific names due to the closed nature of the LWG mailing list). This doesn't look like it will be accepted as P0 Tentatively Ready.

Reply:
We've now spent roughly as much time discussing this as would have been saved by making one change for both the feature and the PR of the LWG issue, So I'll drop this comment.